### PR TITLE
Project Complete

### PR DESCRIPTION
**Pull request by s0kil**
- User Model, Controller, Views
- Event Model, Controller, Views
- EventAttendances Model To Associate User + Event
- Added Unit And Integration Testing

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  # One-To-Many, User (creator) Has Many Events
+  has_many :events
```
Comment (elmejdki) [inline app/models/user.rb]: user this instead: 
`has_many :events, foreign_key: :creator_id`

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      <th>Date</th>
+      <th>Title</th>
+      <th>Location</th>
+      <th>Description</th>
+      <th colspan="3"></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <%= render partial: 'event', collection: Event.upcoming, as: :event %>
```
Comment (elmejdki) [inline app/views/events/index.html.erb]: all code related to accessing data from a database should be inside the model not in the view.

File: app/views/events/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      <th>Date</th>
+      <th>Title</th>
+      <th>Location</th>
+      <th>Description</th>
+      <th colspan="3"></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <%= render partial: 'event', collection: Event.previous, as: :event %>
```
Comment (elmejdki) [inline app/views/events/index.html.erb]: all code related to accessing data from a database should be inside the model not in the view.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
+
+  <body>
+    <%= link_to 'All Events', events_path %>
+    <%= link_to 'Users', users_path %>
+
+    <div class="users-session">
+      <% if current_user %>
```
Comment (elmejdki) [inline app/views/layouts/application.html.erb]: please remove all the logic from your views and put it inside the view helpers `app/helpers/*_helper.rb`

File: app/views/users/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<%= form_with(model: user, local: true) do |form| %>
+  <% if user.errors.any? %>
```
Comment (elmejdki) [inline app/views/users/_form.html.erb]: please remove all the logic from your views and put it inside the view helpers `app/helpers/*_helper.rb`

File: app/views/events/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<%= form_with(model: event, local: true) do |form| %>
+  <% if event.errors.any? %>
```
Comment (elmejdki) [inline app/views/events/_form.html.erb]: please remove all the logic from your views and put it inside the view helpers `app/helpers/*_helper.rb`

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
## Hi, @team great work so far :clap: :clap: :clap: :clap: but you still need to add :toolbox: more to this project to get it approved :checkered_flag: 

- [x] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming

- [x] Navigation links should be present at the top of the page. consider adding some style to your page.

- [x] please remove all the logic from your view, there should be absolutely no logic in view:
image

- [x] Add more tests to your project don't forget to cover the negative and positive tests.

### Please make all the required changes before you submit another coder review.
## Happy Coding :tophat:

File: app/views/events/index.html.erb (hunk 2 above)
Comment (s0kil) [reply app/views/events/index.html.erb]: This does not really click for me, how else if the data supposed to get to the view?

File: app/views/users/_form.html.erb (hunk 5 above)
Comment (eliasecasta) [reply app/views/users/_form.html.erb]: Done

File: app/views/events/_form.html.erb (hunk 6 above)
Comment (eliasecasta) [reply app/views/events/_form.html.erb]: Done

**Review by christianotieno — APPROVED** [review]
# Project Approved 🍾 
Hello team 👋 
Congratulations on meeting the required changes.  👏 
This project has been approved. 🎆 
All the very best in your next tasks. 🥇 

### Happy coding 💻 

#### Christian Otieno (TSE)
